api/src/common/utils.rs

```diff
@@ -23,29 +28,90 @@ pub const fn near_data_to_near_token(data: near_api_types::Data<u128>) -> NearTo
 // We default to false as we can't know if an error is critical or not without the types
 // so to keep it safe it's better to retry
 
-pub fn is_critical_blocks_error(err: &RpcError) -> bool {
-    is_critical_json_rpc_error(err, |_| false)
+pub fn to_retry_error<T, E: std::fmt::Debug + Send + Sync>(
+    err: SendRequestError<E>,
+    is_critical_t: impl Fn(&SendRequestError<E>) -> bool,
+) -> RetryResponse<T, SendRequestError<E>> {
+    if is_critical_t(&err) {
+        RetryResponse::Critical(err)
+    } else {
+        RetryResponse::Retry(err)
+    }
 }
 
-pub fn is_critical_validator_error(err: &RpcError) -> bool {
-    is_critical_json_rpc_error(err, |_| false)
+pub fn is_critical_blocks_error(err: &SendRequestError<RpcBlockError>) -> bool {
+    is_critical_json_rpc_error(err, |err| match err {
+        RpcBlockError::UnknownBlock { .. }
```

Comment:
If it's not one of those, then should be false. To be compatible with newer releases. Please update here and in other places

Reply:
I'm not sure that I'm in favor of _ => false

I would rather fail compilation.
It would be easier to maintain and less prone to unexpected variations that were added...

If the new values are added, it should be a breaking change, I think, so it shouldn't pop up in user code

On the other hand it potentially makes future releases incompatible with older sdk.